### DISCUSSION TITLE
How to trace to the vlan-group based on the interface of a device

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#8938] [Q&A]
Hi

I am having to figure out which vlan-group to take next available vlan from, based on which interface on a device I  am deploying config to. 
Example: 
I Netbox we have documented something like this:
Device1 - interface AE1 (consists of 2 (or more) interfaces) --> connected to device2 (most likely in another rack/location)
VLAN-groups (VG) are defined with scope Site Group covering all relevant sites. We have multiple VG per site because we are running IP-fabric everywhere, which means every port is basically its own VG with 4000vlans, bound together by the VNI (we have a custom field for the VNI ID)
Now, how can I identify which VG I need to tag a vlan in? Only solution I currently see, is to have 1 VG for each interface (AE), but that is a lot of VG! Is there a better way?

This is a result of a question I am getting from my manager, about why I almost hardcode which interface goes to what vlan-group in our scripts. The reason is that I have not come up with a better way yet.

**Comment by martinum4** [comment]
Hello,
I tried the following in the nbshell, it should work elsewhere similar:
```
> iface = Interface.objects.get(id=31337)
> iface.untagged_vlan.group
<VLANGroup: redacted>
```
Does this help you a bit?
